### PR TITLE
cleanup package and reduce exposed interface

### PR DESCRIPTION
**Pull request by bobby-stripe**
This does a few things:
* gets rid of the global `goforit` instance.  We ban it internally at Stripe, the README says we may introduce breaking changes at any point (and I'm not sure this actually has any external users to break).
* only create a `statsd.New` instance if we didn't pass one into the `goforit` constructor (probably benign, but it seems nice to avoid leaving dangling resources that aren't closed)
* Adds `json` tags to the v2 flag structs to ensure we can serialize in-memory flags to disk and read them back (this was my ultimate goal here, to enable better testing).
* split the v1 and v2 flags out into sub packages.  This reduces the module's public API to:

```
package goforit // import "github.com/stripe/goforit"

const DefaultInterval = 30 * time.Second
const DefaultStatsdAddr = "127.0.0.1:8200"
func Override(ctx context.Context, name string, value bool) context.Context
type Backend interface{ ... }
    func BackendFromFile(filename string) Backend
    func BackendFromJSONFile(filename string) Backend
    func BackendFromJSONFile2(filename string) Backend
type Goforit interface{ ... }
    func New(interval time.Duration, backend Backend, opts ...Option) Goforit
type Option interface{ ... }
    func DeletedCallback(cb evaluationCallback) Option
    func EvaluationCallback(cb evaluationCallback) Option
    func Logger(printf func(msg string, args ...interface{})) Option
    func Statsd(stats StatsdClient) Option
type StatsdClient interface{ ... }
```

It might be easier to go commit-by-commit -- the breaking into sub packages makes for a super annoying diff (sorry)

### CODE REVIEW

**Review by idiamond-stripe — APPROVED** [review]
lgtm with one comment

File: goforit.go [hunk 1]
```diff
@@ -97,14 +100,33 @@ type goforit struct {
...
+
 const DefaultInterval = 30 * time.Second
 
 func newWithoutInit(enabledTickerInterval time.Duration) *goforit {
-	stats, _ := statsd.New(DefaultStatsdAddr)
 	return &goforit{
 		flags:                 newFastFlags(),
 		defaultTags:           newFastTags(),
-		stats:                 stats,
+		stats:                 new(nullStatsd),
```
[the remark this answers is not in the document]
Comment (bobby-stripe) [reply goforit.go]: oh great find!

**Comment by bobby-stripe** [comment]
re-r? @idiamond-stripe 

took your suggestion

**Review by idiamond-stripe — APPROVED** [review]

**Comment by vasi-stripe** [comment]
FWIW, `Override()` is only global because of the global goforit. We should probably eventually move to a testingGoforit (that implements Goforit) instead of relyingon context.

**Comment by bobby-stripe** [comment]
@vasi-stripe oh neat!  thanks - I was missing that context

**Pull request merged**